diff --git a/CMakeLists.txt b/CMakeLists.txt
index f11e6e2..b70c952 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -121,13 +121,19 @@ if(CMAKE_VERSION VERSION_GREATER "3.15.0" OR CMAKE_VERSION VERSION_EQUAL "3.15.0
         cmake_policy(SET CMP0094 NEW)
     endif()
     set(_args Interpreter)
-    if (CLINGO_BUILD_WITH_PYTHON)
+    if (CLINGO_BUILD_WITH_PYTHON AND NOT PYCLINGO_DYNAMIC_LOOKUP)
         set(_args ${_args} Development)
         if (CLINGO_REQUIRE_PYTHON)
             set(_args ${_args} REQUIRED)
         endif()
     endif()
     find_package(Python ${CLINGO_PYTHON_VERSION} COMPONENTS ${_args})
+    if (PYCLINGO_DYNAMIC_LOOKUP AND (Python_INCLUDE_DIRS OR PYTHON_INCLUDE_DIR OR Python_INCLUDE_DIR))
+        set(Python_Development_FOUND ON)
+        add_library(Python::Python INTERFACE IMPORTED)
+        add_library(Python::Module INTERFACE IMPORTED)
+        set_property(TARGET Python::Python Python::Module PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${Python_INCLUDE_DIRS} ${PYTHON_INCLUDE_DIR} ${Python_INCLUDE_DIR})
+    endif()
 else()
     find_package(PythonInterp)
     set(Python_EXECUTABLE "${PYTHON_EXECUTABLE}")
@@ -245,4 +251,3 @@ if (NOT CLINGO_BUILD_STATIC AND (CLINGO_BUILD_SHARED OR Python_Development_FOUND
         set_target_properties(${target} PROPERTIES POSITION_INDEPENDENT_CODE ON)
     endforeach()
 endif()
-
diff --git a/app/pyclingo/CMakeLists.txt b/app/pyclingo/CMakeLists.txt
index 22e50f1..056d63b 100644
--- a/app/pyclingo/CMakeLists.txt
+++ b/app/pyclingo/CMakeLists.txt
@@ -32,7 +32,7 @@ if (NOT PYCLINGO_SUFFIX)
 endif()
 
 add_library(pyclingo MODULE ${source})
-if (PYCLINGO_DYNAMIC_LOOKUP)
+if (PYCLINGO_DYNAMIC_LOOKUP AND APPLE)
     set_property(TARGET pyclingo PROPERTY LINK_FLAGS "-undefined dynamic_lookup")
 endif()
 target_link_libraries(pyclingo PRIVATE libpyclingom)
diff --git a/app/pyclingo_cffi/CMakeLists.txt b/app/pyclingo_cffi/CMakeLists.txt
index cd7c8c4..174785e 100644
--- a/app/pyclingo_cffi/CMakeLists.txt
+++ b/app/pyclingo_cffi/CMakeLists.txt
@@ -33,7 +33,7 @@ endif()
 
 add_library(pyclingo MODULE ${source})
 target_link_libraries(pyclingo PRIVATE libclingo Python::Python)
-if (PYCLINGO_DYNAMIC_LOOKUP)
+if (PYCLINGO_DYNAMIC_LOOKUP AND APPLE)
     set_property(TARGET pyclingo PROPERTY LINK_FLAGS "-undefined dynamic_lookup")
 endif()
 set_target_properties(pyclingo PROPERTIES
diff --git a/setup.py b/setup.py
index 76f4582..4616110 100644
--- a/setup.py
+++ b/setup.py
@@ -32,6 +32,7 @@ setup(
                  '-DCLINGO_BUILD_STATIC=OFF',
                  '-DCLINGO_BUILD_SHARED=OFF',
                  '-DCLINGO_REQUIRE_PYTHON=ON',
+                 '-DPYCLINGO_DYNAMIC_LOOKUP=ON',
                  '-DCLINGO_BUILD_WITH_LUA=OFF',
                  '-DPYCLINGO_USE_CFFI=ON',
                  '-DPYCLINGO_USER_INSTALL=OFF',
